firebase-firestore/src/test/java/com/google/firebase/firestore/local/LocalStoreTestCase.java

```diff
@@ -934,4 +949,143 @@ public void testRemoteDocumentKeysForTarget() {
     keys = localStore.getRemoteDocumentKeys(2);
     assertSetEquals(asList(key("foo/bar"), key("foo/baz")), keys);
   }
+
+  @Test
+  public void testHandlesSetMutationThenTransformMutationThenTransformMutation() {
+    writeMutation(setMutation("foo/bar", map("sum", 0)));
+    assertContains(doc("foo/bar", 0, map("sum", 0), Document.DocumentState.LOCAL_MUTATIONS));
+    assertChanged(doc("foo/bar", 0, map("sum", 0), Document.DocumentState.LOCAL_MUTATIONS));
+
+    writeMutation(transformMutation("foo/bar", map("sum", FieldValue.numericAdd(1))));
+    assertContains(doc("foo/bar", 0, map("sum", 1), Document.DocumentState.LOCAL_MUTATIONS));
+    assertChanged(doc("foo/bar", 0, map("sum", 1), Document.DocumentState.LOCAL_MUTATIONS));
+
+    writeMutation(transformMutation("foo/bar", map("sum", FieldValue.numericAdd(2))));
+    assertContains(doc("foo/bar", 0, map("sum", 3), Document.DocumentState.LOCAL_MUTATIONS));
+    assertChanged(doc("foo/bar", 0, map("sum", 3), Document.DocumentState.LOCAL_MUTATIONS));
+  }
+
+  @Test
+  public void testHandlesSetMutationThenAckThenTransformMutationThenAckThenTransformMutation() {
```

Comment:
Did you consider what it would take to be able to make these tests spec tests?  I think that'd be preferable but I think spec tests don't currently do transforms? :-/

Reply:
Maybe add:

// TODO(mrschmidt): These tests would probably be better as spec tests but currently they don't support transforms.

or something?

Reply:
Done